iml-manager-cli/src/filesystem.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    api_utils::{get, get_all, get_one},
+    display_utils::{generate_table, wrap_fut},
+    error::ImlManagerCliError,
+};
+use iml_wire_types::{ApiList, Filesystem, FlatQuery, Mgt};
+use number_formatter::{format_bytes, format_number};
+use prettytable::{Row, Table};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub enum FilesystemCommand {
+    /// List all configured filesystems
+    #[structopt(name = "list")]
+    List,
+    /// Show filesystem
+    #[structopt(name = "show")]
+    Show {
+        #[structopt(name = "FSNAME")]
+        fsname: String,
+    },
+}
+
+fn usage(
+    free: Option<f64>,
+    total: Option<f64>,
+    formatter: fn(f64, Option<usize>) -> String,
+) -> String {
+    match (free, total) {
+        (Some(free), Some(total)) => format!(
+            "{} / {}",
+            formatter(total - free, Some(0)),
+            formatter(total, Some(0))
+        ),
+        (None, Some(total)) => format!("Calculating ... / {}", formatter(total, Some(0))),
+        _ => "Calculating ...".to_string(),
+    }
+}
+
+pub async fn filesystem_cli(command: FilesystemCommand) -> Result<(), ImlManagerCliError> {
+    match command {
+        FilesystemCommand::List => {
+            let filesystems: ApiList<Filesystem> =
+                wrap_fut("Fetching filesystems...", get_all()).await?;
+
+            tracing::debug!("FSs: {:?}", filesystems);
+
+            let table = generate_table(
+                &[
+                    "Name", "State", "Space", "Inodes", "Clients", "MDTs", "OSTs",
+                ],
+                filesystems.objects.into_iter().map(|f| {
+                    vec![
+                        f.label,
+                        f.state,
+                        usage(f.bytes_free, f.bytes_total, format_bytes),
+                        usage(f.files_free, f.files_total, format_number),
+                        format_number(f.client_count.unwrap_or(0.0), Some(0)),
+                        f.mdts.len().to_string(),
```

Comment:
Am I missing something, or will just list the length of the mdts resource_uri string?

Don't you need a separate fetch to get the mdts based on the resource URI?

Reply:
MDTs come through as Vec:
https://github.com/whamcloud/integrated-manager-for-lustre/blob/master/iml-wire-types/wire-types.rs#L771